jaeger/cmd/root.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultLinkerdNamespace = "linkerd"
+)
+
+var (
+	apiAddr               string // An empty value means "use the Kubernetes configuration"
+	controlPlaneNamespace string
+	kubeconfigPath        string
+	kubeContext           string
+	impersonate           string
+	impersonateGroup      []string
+	verbose               bool
+
+	// These regexs are not as strict as they could be, but are a quick and dirty
+	// sanity check against illegal characters.
+	alphaNumDash = regexp.MustCompile(`^[a-zA-Z0-9-]+$`)
+)
+
+// RootCmd represents the root Cobra command
+var RootCmd = &cobra.Command{
+	Use:   "jaeger",
+	Short: "jaeger manages the jaeger extension of Linkerd service mesh",
+	Long:  `jaeger manages the jaeger extension of Linkerd service mesh.`,
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		// enable / disable logging
+		if verbose {
+			log.SetLevel(log.DebugLevel)
+		} else {
+			log.SetLevel(log.PanicLevel)
+		}
+
+		controlPlaneNamespaceFromEnv := os.Getenv("LINKERD_NAMESPACE")
```

Comment:
Why do we need to read the linkerd namespace from an env variable in addition to the flag?

Reply:
This follows the same logic of Linkerd, where users can also set the linkerd namespace using env and hence our binary also should honor that?

Reply:
Not sure why that's there in the main CLI either, but I don't think we need to copy that behavior.  Let's not do stuff unless we know why.